stdlib/Random/src/ppmisc.jl

```diff
@@ -0,0 +1,167 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+#
+# Parallel Partitioned Shuffle
+#
+using Base.Threads
+export ppshuffle!, ppshuffle, pprandperm!, pprandperm
+
+## ppshuffle! & ppshuffle
+
+"""
+    _ppshuffle!(rng::TaskLocalRNG, B::AbstractArray{T}, A::Union{AbstractArray{T}, Base.OneTo{T}}, mask<:Union{UInt8, UInt16})
+
+Parallel Partitioned Shuffle
+1. partition input randomly
+2. shuffle partitions concurrently
+
+Arg `mask` determines number of partitions (mask + 1) to be used.
+"""
+function _ppshuffle!(r::TaskLocalRNG, B::AbstractArray{T}, A::Union{AbstractArray{T}, Base.OneTo{T}}, mask::Tu) where {T, Tu<:Union{UInt8, UInt16}}
```

Comment:
`Base.OneTo{T}` already is an `AbstractArray` so this union isn't doing anything. Also should it be `AbstractVector` rather than `AbstractArray`?

Reply:
> `Base.OneTo{T}` already is an `AbstractArray` so this union isn't doing anything. 

It is a valid point - this union is redundant. I will correct it.

> Also should it be `AbstractVector` rather than `AbstractArray`?

I simply mimicked the signature of the default method.
```julia 
function shuffle!(r::AbstractRNG, a::AbstractArray)
```

Reply:
see 175bad771c85ec5dbfb3d1ff0c59a615332deeb4